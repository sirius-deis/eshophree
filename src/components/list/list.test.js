import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import List from './list';

describe('List component', () => {
  const list = ['1', '2', '3'];
  it('should match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <List list={list} />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render component with children', () => {
    render(
      <MemoryRouter>
        <List list={list} />
      </MemoryRouter>,
    );
    const content = screen.queryAllByRole('listitem');
    expect(content.length).toBe(3);
    expect(content[0].textContent).toBe('1');
    expect(content[1].textContent).toBe('2');
    expect(content[2].textContent).toBe('3');
  });
});
