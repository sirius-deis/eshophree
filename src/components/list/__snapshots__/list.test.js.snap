// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should match snapshot 1`] = `
<div>
  <ul
    class="sc-bgqQcB eLjcYi"
  >
    <li
      class="sc-gTRrQi kQsCbO"
    >
      <a
        href="/1"
      >
        1
      </a>
    </li>
    <li
      class="sc-gTRrQi kQsCbO"
    >
      <a
        href="/2"
      >
        2
      </a>
    </li>
    <li
      class="sc-gTRrQi kQsCbO"
    >
      <a
        href="/3"
      >
        3
      </a>
    </li>
  </ul>
</div>
`;
